File: CS50x-Week1-Solution-With-Deep-Explanation-main/mario (feeling more comfortable)/mario.c
#include <stdio.h> // Include the standard input/output library for printing output
#include <cs50.h> // Include the CS50 library for getting input from the user

int main(void)
{
    int height; // Variable to store the height of the object (pyramid)

    // Prompt the user to enter a height between 1 and 8 (inclusive)
    do
    {
        height = get_int("Write the height of an object: "); // Get the height from the user
    }
    while (height < 1 || height > 8); // Repeat the prompt if the height is not within the valid range

    // Loop to construct each row of the pyramid
    for (int i = 1; i <= height; i++)
    {
        // Print spaces before the hashes for the left side of the pyramid
        for (int j = 0; j < height - i; j++)
        {
            printf(" "); // Print a space
        }

        // Print hashes for the left side of the pyramid
        for (int j = 0; j < i; j++)
        {
            printf("#"); // Print a hash
        }

        // Print two spaces between the two sides of the pyramid
        printf("  "); // Print two spaces

        // Print hashes for the right side of the pyramid
        for (int j = 0; j < i; j++)
        {
            printf("#"); // Print a hash
        }

        // Move to the next line after completing the current row
        printf("\n");
    }

    return 0; // Return 0 to indicate successful completion of the program
}

File: CS50x-Week6-Solution-With-Deep-Explanation-main/mario (feeling more comfortable)/mario.py
from cs50 import get_int  # Import the get_int function from the cs50 module for integer input

height = 0  # Initialize height to 0 (or any value outside the valid range)

# Continuously prompt the user to enter a valid height (between 1 and 8 inclusive)
while height < 1 or height > 8:
    height = get_int("Height: ")  # Read an integer input from the user and store it in 'height'

# Generate and print the pyramid
for i in range(1, height + 1):
    # Construct and print each row of the pyramid
    # ' ' * (height - i) calculates the number of leading spaces needed for the left-aligned portion
    # '#' * i creates the number of hashes needed for the left-aligned portion of the pyramid
    # '  ' adds two spaces between the two pyramid halves
    # '#' * i creates the number of hashes needed for the right-aligned portion of the pyramid
    # 'sep=""' prevents adding extra spaces between the concatenated strings
    print(" " * (height - i) + "#" * i + "  " + "#" * i, sep="")
